packages/run-protocol/src/feeDistributor.js

```diff
@@ -0,0 +1,308 @@
+// @ts-check
+
+import { AmountMath } from '@agoric/ertp';
+import { E, Far } from '@endo/far';
+import { observeNotifier } from '@agoric/notifier';
+import { makeScalarSetStore } from '@agoric/store';
+
+/**
+ * @typedef {object} FeeCollector
+ *
+ * @property {() => ERef<Payment>} collectFees
+ */
+
+/**
+ * @typedef {object} PeriodicFeeCollector
+ * @property {() => FeeCollector} getCollector
+ * @property {() => Promise<void>} collectAndDistributeNow
+ * @property {() => void} stop
+ */
+
+/**
+ * @typedef {object} CollectibleContractFacet
+ * @property {() => Promise<Invitation>} makeCollectFeesInvitation
+ */
+
+/**
+ * wrapper to take the vaultFactory or AMM's creatorFacet, and make a function that
+ * will request an invitation and return a promise for a payment.
+ *
+ * @param {ERef<ZoeService>} zoe
+ * @param {ERef<CollectibleContractFacet>} creatorFacet
+ */
+export function makeContractFeeCollector(zoe, creatorFacet) {
+  /** @type {FeeCollector} */
+  return Far('feeCollector', {
+    collectFees: () => {
+      const invitation = E(creatorFacet).makeCollectFeesInvitation();
+      const collectFeesSeat = E(zoe).offer(invitation, undefined, undefined);
+      return E(collectFeesSeat).getPayout('RUN');
+    },
+  });
+}
+
+/**
+ * A distributor of fees from Inter Protocol sources to deposit facets. Each
+ * time the epochTimer signals the end of an Epoch, it will ask the contracts
+ * for fees that have been collected to date and send that payment to the
+ * depositFacet.
+ *
+ * @param {() => Promise<unknown>} schedulePayments - distribute to the destinations
+ * @param {ERef<TimerService>} timerService - timer that is used to schedule collections
+ * @param {RelativeTime} [collectionInterval] - how often to collect fees in the
+ * `timerService` unit
+ */
+export function startDistributing(
```

Comment:
same as above, not imported and should be const

Reply:
Agreed on constness.  I'll leave it for future `function` keyword cleanups that may be more extensive.